tests/unit/test_errorfactory.py

```diff
@@ -0,0 +1,30 @@
+import sys
+
+from nose.tools import assert_equals, assert_true
+
+from botocore.exceptions import ClientError
+from botocore.errorfactory import ServiceErrorFactory
+from botocore.model import ServiceModel
+
+
+def test_errorfactory():
+    model = {
+        'operations': {
+            'OperationName': {
+                'name': 'OperationName',
+                'errors': [{'shape': 'NoSuchResourceException'}],
+            }
+        },
+        'shapes': {
+            'NoSuchResourceException': {
+                'type': 'structure',
+                'members': {}
+            }
+        }
+    }
+    factory = ServiceErrorFactory(ServiceModel(model))
+    members = dir(factory)
+    if sys.version_info >= (2, 7):
```

Comment:
This seems a little fishy. Ideally the tests should be able to handle any python version we support, and if there is no way to support the test then we skip it. I saw your note of ``dir()`` not working on Python2.6. What did ``dir()`` return instead? It would be great if we can figure out why there are issues with it.